### ISSUE TITLE
[SAVESTATES] start with --entryslot may need some improovements  

### ISSUE THREAD

**Issue by gugueU** [libretro/RetroArch#14780]
Start with  xxxx.state.entry  or xxxx.state0.entry do not works

Could we update actual selected slot after start a game.
exemple: retroarh --entryslot 5  does update selected slot to 5

**Comment by gugueU** [comment]
This feat seems to be appear with this commit
https://github.com/libretro/RetroArch/commit/682bbab233ab1612eb69abf36ac9e0bbcbbfc1a1
thanks to  @asciibeats :+1: 

I was waiting for longtime the possibility

**Comment by gugueU** [comment]
may be add a thing ike this
 `configuration_set_int(settings, settings->ints.state_slot, runloop_st->entry_state_slot);`

**Comment by gugueU** [comment]
I made a PR which works for me

**Comment by JoeOsborn** [comment]
I think this is fixed and should be closed, right?

**Comment by gugueU** [comment]
Since I opened the issue, we can now load xxxx.state.entry (slot 0) and the lot number will be updated ?

**Comment by gugueU** [comment]
May be need some improvements like:
- allowed to load directly a state, not an entry copy file
- load state number 0

**Comment by gugueU** [comment]
I have got a patch WIP, the code is not clean because of unsigned int and slot 0

**Comment by JoeOsborn** [comment]
For the first one, do you mean allowing the command line argument to be either a number or a path?

And for the second, are you suggesting removing the idea of an “entry state” and just loading the state in the given slot number?

**Comment by gugueU** [comment]
1. yes do not load a copy of a state renamed with .entry. Why do n load directly the original state file.
2. actually we cannot load -entrySlot 0,  because 0 not allowed with unsigned type when we check it after string conversion.
0 or unauthorized string give same results => invalid slot number

**Comment by JoeOsborn** [comment]
Makes sense, I don’t know if anyone is relying on entrySlot besides me and it seems safe to change in this way, but I’ll wait for a maintainer to chime in.  To me it makes sense to eliminate the .entry stuff and just use regular state slots.

0 seems like it could be safe to parse with an additional check besides strtoul like “the first character is literal 0”. But I don’t feel strongly about whether 0 should be .state or .state0, both have downsides.  I find entryslot a bit confusing, but I tried to make minimal patches earlier.

**Comment by gugueU** [comment]
In my patch  at the begining I convet real unsigned 0 to 9999 (non used slot number) and re convert it to 0 before get state path. But I prefere a cleaner way to do that. 
why I do that ? because often we check if(runloop_st->entry_state_slot) 

I also remote the .entry for **retroarch_get_entry_state_path** method
